web/cobrands/buckinghamshire/assets.js

```diff
@@ -169,14 +173,43 @@ fixmystreet.assets.add(labeled_defaults, {
 
 fixmystreet.assets.add(defaults, {
     http_options: {
+        url: proxy_base_url + 'drains/wfs',
         params: {
-            TYPENAME: "Gullies"
+            TYPENAME: "junctions"
         }
     },
-    asset_category: [
-        'Blocked drain'
-        ],
-    asset_item: 'drain'
+    asset_category: ["Blocked drain"],
+    asset_item: 'drain',
+    select_action: true,
+    construct_selected_asset_message: function(asset) {
+        var last_clean = '';
+        if (asset.attributes.last_inspected && asset.attributes.cleaned === 'Yes') {
+            last_clean = munge_date(asset.attributes.last_inspected);
+        }
+        var next_clean = munge_date(asset.attributes.cyclical_target_date);
+        var message = '';
+            if (last_clean != '') { message += 'This gulley was last cleaned on ' + last_clean; }
+            if (next_clean != '' && last_clean != '') {
+                message += '. It will be cleaned again before ' + next_clean;
+            } else if (next_clean != '') {
+                message += 'This gulley will be cleaned before ' + next_clean;
+            }
+        return message || "This gulley has no cleaning history";
+
+        function munge_date(date_field) {
+            var RegExDate = /([0-9]{4})-([0-9]{2})-([0-9]{2})/;
+            var myMatch = RegExDate.exec(date_field);
+            if (myMatch) {
+                return myMatch[3] + '/' + myMatch[2] + '/' + myMatch[1];
+            } else {
+                return '';
+            }
+        }
```

Comment:
I'm not sure if we have a specific policy on this, but I'd prefer this `munge_date` function to be at the top level, rather than nested inside the calling function.